Extension/cpp_snippets.json

```diff
@@ -0,0 +1,351 @@
+//////////////////////////////// 
+// class_snippets
+{
+  "class": {
+    "prefix": "class",
+    "body": "\nclass ${1:${TM_FILENAME_BASE}}\n{\nprivate:\n\t${2:/* data */}\npublic:\n\t${1}(${3:/* args */});\n\t~${1}();\n};\n\n${1}::${1}(${3})\n{\n}\n\n${1}::~${1}()\n{\n}\n",
+    "description": "Code snippet for class",
+    "scope": "source.c++, source.objc++"
+  },
+  "classi": {
+    "prefix": "classi",
+    "body": "\nclass ${1:${TM_FILENAME_BASE}}\n{\nprivate:\n\t${2:/* data */}\npublic:\n\t${1}::${1}(${3:/* args */}) { $0 }\n\t${1}::~${1}() { }\n};",
+    "description": "Code snippet for class with inlined constructor/destructor",
+    "scope": "source.c++, source.objc++"
+  },
+  "classt": {
+    "prefix": "classt",
+    "body": "\ntemplate<typename T}>\nclass ${1:${TM_FILENAME_BASE}}\n{\nprivate:\n\tT ${2:/* data */}\npublic:\n\t${1}(T ${3:/* args */});\n};\n",
+    "description": "Code snippet for class template",
+    "scope": "source.c++, source.objc++"
+  },
+  "excep": {
+    "prefix": "excep",
+    "body": "\nclass ${1:${TM_FILENAME_BASE}} : public ${2:std::runtime_error}\n{\npublic:\n\t${1}(const std::string& str): ${2:std::runtime_error}(${3:str}) {}\n};\n",
+    "description": "Code snippet for exception",
+    "scope": "source.c++, source.objc++"
+  },
+  "struct": {
+    "prefix": "struct",
+    "body": "\nstruct ${1:${TM_FILENAME_BASE}}\n{\n\t${0:/* data */}\n};\n",
+    "description": "Code snippet for struct",
+    "scope": "source.c, source.objc, source.c++, source.objc++"
+  },
+  "union": {
+    "prefix": "union",
+    "body": "\nunion ${1:${TM_FILENAME_BASE}}\n{\n\t${0:/* data */}\n};\n",
+    "description": "Code snippet for union",
+    "scope": "source.c, source.objc, source.c++, source.objc++"
+  },
+  "structtemp": {
+    "prefix": "structtemp",
+    "body": "\ntemplate<typename T>\nstruct ${1:${TM_FILENAME_BASE}} {\n\tT ${0:/* data */}\n};\n",
+    "description": "Code snippet for templated Struct",
+    "scope": "source.c++, source.objc++"
+  },
+
+  ////////////////////////////////
+  // conditional_snippets
+
+  "if": {
+    "prefix": "if",
+    "body": "\nif(${1:/* condition */}) {\n\t${0:/* code */}\n}\n",
```

Comment:
nit: Space between keyword and parenthesis

Reply:
Fixed.